esphome/components/scd4x/scd4x.cpp

```diff
@@ -150,6 +156,20 @@ void SCD4XComponent::update() {
     return;
   }
 
+  if (this->pressure_sensor_ != nullptr) {
+    float pressure = this->pressure_sensor_->state / 1000.0;
+    if (!std::isnan(pressure)) {
+      set_ambient_pressure_compensation(this->pressure_sensor_->state / 1000.0);
+    }
+  } else {
+    if (this->altidute_sensor_ != nullptr) {
+      float altidute = this->altidute_sensor_->state;
+      if (!std::isnan(altidute)) {
+        set_ambient_pressure_compensation(this->altidute_sensor_->state);
```

Comment:
```suggestion
        set_altitude_compensation(this->altidute_sensor_->state);
```
(did you test this?)